.deployment/templates/nginx.conf

```
@@ -0,0 +1,98 @@
+# Defines user and group credentials used by worker processes. If group is
+# omitted, a group whose name equals that of user is used.
+user  nginx;
+
+# Defines the number of worker processes.  Setting it to the number of
+# available CPU cores should be a good start. The value `auto` will try to
+# autodetect that.
+worker_processes auto;
+
+# Configures logging to `/var/log/...`. Log level `error` is used by default.
+error_log  /var/log/nginx/error.log;
+
+# Defines a file that will store the process ID of the main process. This needs
+# to match the Systemd unit file.
+pid /run/nginx.pid;
+
+# Load dynamic modules. See /usr/share/nginx/README.dynamic.
+include /usr/share/nginx/modules/*.conf;
+
+events {
+   # Sets the maximum number of simultaneous connections that can be opened by
+   # a worker process.
+   worker_connections  1024;
+}
+
+
+http {
+   # Include mime types for different file extensions.
+   include       /etc/nginx/mime.types;
+
+   # Defines the default MIME type of a response.
+   default_type  application/octet-stream;
+
+   # Sendfile copies data between one file descriptor and other from within the
+   # kernel. This is more efficient than read() and write() since they require
+   # transferring data to and from the user space.
+   sendfile      on;
+
+   # Todo: Write explanation
+   # https://t37.net/nginx-optimization-understanding-sendfile-tcp_nodelay-and-tcp_nopush.html
+   tcp_nopush on;
+   tcp_nodelay on;
+
+   # Disable access log.
+   access_log off;
+
+   # Enable on-the-fly gzip compression for larger plain text files and for
+   # proxies applications.
+   gzip             on;
+   gzip_comp_level  2;
+   gzip_min_length  1000;
+   gzip_proxied     expired no-cache no-store private auth;
+   gzip_types       text/plain application/x-javascript text/xml text/css application/xml;
```

```suggestion
   gzip_types
        application/javascript
        application/json
        application/x-javascript
        application/xml
        image/svg+xml
        text/css
        text/javascript
        text/js
        text/plain
        text/xml;
```

Why is the indentation 3 spaces btw? :D